templates/nft-template/move/sources/nft_launchpad.move

```diff
@@ -0,0 +1,466 @@
+module launchpad_addr::nft_launchpad {
+    use std::option;
+    use std::signer;
+    use std::string;
+    use std::vector;
+    use aptos_std::simple_map;
+    use aptos_std::string_utils;
+    use aptos_framework::aptos_account;
+    use aptos_framework::event;
+    use aptos_framework::object;
+    use aptos_token_objects::collection;
+    use aptos_token_objects::royalty;
+    use aptos_token_objects::token;
+
+    use minter::token_components;
+    use minter::mint_stage;
+    use minter::collection_components;
+
+    /// Sender is not admin
+    const E_NOT_ADMIN: u64 = 1;
+    /// Sender is not pending admin
+    const E_NOT_PENDING_ADMIN: u64 = 2;
+    /// Only admin can update mint fee collector
+    const E_ONLY_ADMIN_CAN_UPDATE_MINT_FEE_COLLECTOR: u64 = 3;
+    /// Only admin can create collection
+    const E_ONLY_ADMIN_CAN_CREATE_COLLECTION: u64 = 4;
+    /// No mint limit
+    const E_NO_MINT_LIMIT: u64 = 5;
+    /// Mint limit reached
+    const E_MINT_LIMIT_REACHED: u64 = 6;
+    /// No active mint stages
+    const E_NO_ACTIVE_STAGES: u64 = 7;
+
+    const ALLOWLIST_MINT_STAGE_CATEGORY: vector<u8> = b"Allowlist mint stage";
+    const PUBLIC_MINT_MINT_STAGE_CATEGORY: vector<u8> = b"Public mint mint stage";
+
+    #[event]
+    struct CreateCollectionEvent has store, drop {
```

Comment:
Do we not have a standard in TokenMinter for this @JohnChangUK? Custom event at this level means that the user has to have indexing to get the full benefit of the example contract. That's exactly what we're trying to avoid with the token standards.

Reply:
No event for collection creation, as collection creation is initially done via the framework. TokenMinter creates the peripherals for the collection only so far.